### DISCUSSION TITLE
Can we add `static` keyword in getters and setters of a skill?

### DISCUSSION THREAD

**Discussion by minhduc0711** [gama-platform/gama#105] [Q&A] [answered]
e.g:
```
@getter (IKeyword.REAL_SPEED)
public static double getRealSpeed(final IAgent agent) {
  if (agent == null) { return 0.0; }
  return (Double) agent.getAttribute(IKeyword.REAL_SPEED);
}
```
I wonder if doing this breaks anything. 
The getters and setters are associated with the passed-in agent anyway, and making it static will allow accessing/modifying attributes among different skills (e.g. accessing a road's speed limit from driving skill).

**Comment by AlexisDrogoul** [comment, marked as answer]
Hi,

Skills are stateless instances in GAMA, so almost the exact equivalent of static classes. The annotation processor is smart enough, also, to call dynamic and static methods in the same way. So no problem. 

Alexis
